engine-rest/src/main/java/org/camunda/bpm/engine/rest/sub/history/impl/HistoricProcessInstanceResourceImpl.java

```diff
@@ -43,4 +46,16 @@ public HistoricProcessInstanceDto getHistoricProcessInstance() {
     return HistoricProcessInstanceDto.fromHistoricProcessInstance(instance);
   }
 
+  @Override
+  public void deleteHistoricProcessInstance() {
+    HistoryService historyService = engine.getHistoryService();
+    try {
+      historyService.deleteHistoricProcessInstance(processInstanceId);
+    } catch (AuthorizationException e) {
+      throw e;
+    } catch (ProcessEngineException e) {
+      throw new InvalidRequestException(Status.NOT_FOUND, e, "Historic process instance with id " + processInstanceId + " does not exist");
```

Comment:
We should probably provide two different messages for the cases when:
- neither `HistoricProcessInstance` nor `ProcessInstance` with the given ID exist
- `ProcessInstance` with the given ID exists (the instance has not yet been moved to history)

`DeleteHistoricProcessInstanceCmd` differentiates between those two cases but throws the same exception for both of them and I didn't want to play with exception message parsing here.